### DISCUSSION TITLE
Referencing CoreWCF.NetTcp.1.5.1 nuget in .net Framework 4.8 project

### DISCUSSION THREAD

**Discussion by AlleSchonWeg** [CoreWCF/CoreWCF#1352] [Q&A]
Hi,
we have a very big .net Framework application. It includes windows services, class librarys and a WebForm app. Two of the windows services communicate via WCF (tcp). Our goal is to migrate to .Net 8 or above.
I added the CoreWCF.NetTcp.1.5.1 nuget to one of the services. The package added almost 100 of other nugets. Is this expected? I also have over 50 updates for the packages. Some of these are deprecated. Microsoft.AspNetCore.Authentication for example.
Is this normal? It's possible to update most of the nugets via package manager to the most recent versions?

Thank you.

**Comment by mconnew** [comment]
You need to either specify the `Web` sdk by having this as the fist line in your project:
```xml
<Project Sdk="Microsoft.NET.Sdk.Web">
```
Or you need to include this in your project:
```xml
  <ItemGroup>
    <FrameworkReference Include="Microsoft.AspNetCore.App" />
  </ItemGroup>
```

All the packages being pulled in are what's needed for when using asp.net core 2.1 with .NET Framework. By specifying the `Web` sdk or adding the `FrameworkReference`, it adds some pieces to the build which removes all those transitive dependencies if they have been replaced by the in runtime aspnetcore implementation.  

We purposefully reference the oldest LTS still in support versions of packages (latest patch level, but oldest supported major version) to avoid upgrading dependencies or causing build warnings that you've downgraded something that CoreWCF wants a later version of. You might see packages saying they can be upgraded because there are now 8.x versions of the packages available. You can upgrade them or choose to leave them at the 6.x versions as both are supported. Around the time that .net 6.0 goes end of life, we'll release a version of CoreWCF which depends on the 8.x versions of the packages.
